config/config_helpers.go

```diff
@@ -524,13 +524,20 @@ func sopsDecryptFile(params []string, include *IncludeConfig, terragruntOptions
 		return "", errors.WithStackTrace(InvalidSopsFormat{SourceFilePath: sourceFile})
 	}
 
+	if val, ok := SOPSCACHE[sourceFile]; ok {
```

Comment:
Hm, we may want to use a canonical path here... That way, if different modules specify the path to the same file, but in different ways (e.g., `../../foo` in one place and `/x/y/foo` in another), they both still get the same cache hit.

Try using [`util.CanonicalPath`](https://github.com/gruntwork-io/terragrunt/blob/master/util/file.go#L33) with `terragruntOptions.WorkingDir` as the `basePath`.